src/services/api/routes/community.ts

```diff
@@ -122,14 +126,92 @@ class ApiRouteCommunity {
     }
 
     static async create(
+        uri: string,
         details: CommunityCreationAttributes
     ): Promise<{ data: CommunityAttributes; error: any }> {
+        const mimetype = mime
+            .contentType(path.basename(uri))
+            .match(/\/(\w+);?/)[1];
+        const preSigned = (
+            await this.api.get<{ uploadURL: string; media: AppMediaContent }>(
+                '/community/media/' + mimetype,
+                true
+            )
+        ).data;
+        const ru = await FileSystem.uploadAsync(preSigned.uploadURL, uri, {
+            httpMethod: 'PUT',
+            mimeType: mimetype,
+            uploadType: 0, //FileSystemUploadType.BINARY_CONTENT
+            headers: {
+                'Content-Type': 'image/' + mimetype,
+            },
+        });
+        if (ru.status >= 400) {
+            throw new Error(ru.body.toString());
+        }
```

Comment:
Is there any error treatment/handler when throwing this error?

Reply:
Yes.
The whole process is surrounded by try/catch. https://github.com/impactMarket/mobile-app/pull/112/files/e2c8bbb8a91ed6606dd2d58dd46ff5166c082726#diff-37cf8af5e15d4449d5297985e2c1e148f148b90af3ad2b2e0c20cb326424f742L694